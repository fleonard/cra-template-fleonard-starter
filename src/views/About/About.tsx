import React, { Fragment } from "react";
import { useHistory } from "react-router-dom";
import { Button, Typography } from "@material-ui/core";

const About: React.FC = () => {
  const history = useHistory();

  return (
    <Fragment>
      <Typography variant="h1">About</Typography>
      <Button
        variant="contained"
        color="primary"
        type="button"
        className="btn"
        cy-data="go-back-button"
        onClick={() => history.push("/")}
      >
        Go back
      </Button>
    </Fragment>
  );
};

export default About;
