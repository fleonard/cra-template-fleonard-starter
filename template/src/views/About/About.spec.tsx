import React from "react";
import { render } from "@testing-library/react";
import "jest-styled-components";

import About from "./About";

describe("About Component", () => {
  it("renders correctly", () => {
    const { container } = render(<About />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
