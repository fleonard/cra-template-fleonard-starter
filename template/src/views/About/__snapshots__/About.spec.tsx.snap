// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Component renders correctly 1`] = `
<h1
  class="MuiTypography-root MuiTypography-h1"
>
  About
</h1>
`;
